### PR TITLE
SNOW-1284606 Move running_on_* to tests/utils.py

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
Those running_on_* methods was in `tests/integ/conftest.py` which requires `tests.parameters` available. It blocks us to use them in sproc tests since there is no `tests.parameters` there.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -89,6 +90,22 @@
 ICEBERG_ENVIRONMENTS = {"dev", "aws"}
 IS_ICEBERG_SUPPORTED = os.getenv("cloud_provider", "dev") in ICEBERG_ENVIRONMENTS
 
+RUNNING_ON_GH = os.getenv("GITHUB_ACTIONS") == "true"
+RUNNING_ON_JENKINS = "JENKINS_HOME" in os.environ
+TEST_SCHEMA = f"GH_JOB_{(str(uuid.uuid4()).replace('-', '_'))}"
```
Comment (sfc-gh-lmukhopadhyay) [inline tests/utils.py:95]: nit - can we add this?
```suggestion
if RUNNING_ON_GH:
    TEST_SCHEMA = f"GH_JOB_{(str(uuid.uuid4()).replace('-', '_'))}"
```

**Review by sfc-gh-lmukhopadhyay — COMMENTED** [review]
LGTM thanks!

File: tests/utils.py (hunk 1 above)
Comment (sfc-gh-azhan) [reply tests/utils.py:95]: that one also used as default one

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM!

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]

**Comment by sfc-gh-azhan** [comment]
We are not showing any implemented APIs from src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py in our doc. Can you help to fix it too?

**Review by sfc-gh-jdu — APPROVED** [review]

**Pull request merged**